### PR TITLE
Add pre-build hooks

### LINKED ISSUE #170: Add features to documentation
There are some features and tweaks that need to be documented after #166:
* Page Types
* Global Pre-Build Hooks
* Theme Pre-Build Hooks
* Paths for generated assets ignored by the watcher: `static/generated`
* Changed theme directory structure once #169 is implemented

### PR DESCRIPTION
**Pull request by dominikbraun**
Fixes #170.

### CODE REVIEW

File: docs/theme-reference.md [hunk 1]
```diff
@@ -5,33 +5,62 @@ you therefore need to customize the default theme, create your own theme or use
...
+
+```shell script
+$ verless create theme my-blog dark-theme
+```
+
+Or, if you already are _inside_ the `my-blog` directory:
+
+```shell script
+$ verless create theme . dark-theme
+```
```
Comment (hmble) [inline docs/theme-reference.md:39]: We don't need to pass `.` to this command as it is default project path. The command should be
`verless create theme dark-theme`

Comment (dominikbraun) [reply docs/theme-reference.md:39]: You're right, this needs to be updated. Thanks 👍

File: docs/theme-reference.md [hunk 2]
```diff
@@ -5,33 +5,62 @@ you therefore need to customize the default theme, create your own theme or use
...
 
 ```shell script
 my-blog/
 └── themes/
     └── dark-theme/
-        ├── css/
-        │   └── style.css
-        ├── js/
+        ├── theme.yml
+        ├── assets/
```
Comment (aligator) [inline docs/theme-reference.md:53]:
```suggestion
        ├── assets/
        ├── generated/ (optional)
```

**Review by aligator — APPROVED** [review]
LGTM

**Review by aligator — APPROVED** [review]
LGTM

**Pull request merged**